build/android/src/main/java/net.minetest.minetest/AssetCopyTask.java

```diff
@@ -0,0 +1,289 @@
+package net.minetest.minetest;
+
+import android.content.Intent;
+import android.content.res.AssetFileDescriptor;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.os.Environment;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.util.Vector;
+
+class AssetCopyTask extends AsyncTask<String, Integer, String> {
+	private static final String TAG = "AssetCopyTask";
+	private static final String gameAssetPrefix = "files" + File.separator;
+
+	private File basePath;
+	private AssetCopyActivity activity;
+
+	private boolean copyStarted = false;
+	private String folderName = "media";
+
+	private Vector<String> folderNames = new Vector<>();
```

Comment:
I knew about the different reallocation strategies, but I missed the synchronization part.  This should be changed.